### PR TITLE
fix: Removed ejs for manifest updates

### PR DESCRIPTION
**Pull request by broksy**
As `manifest.json` template started to look more and more complex and test coverage was difficult to check, I removed ejs in manifest templating and wrote ts functions for enhancing manifest.

### CODE REVIEW

**Review by devinea — APPROVED** [review]

File: packages/odata-service-writer/templates/extend/manifest.json [hunk 1]
```diff
@@ -57,7 +57,7 @@
             <% } %>
             <% if (locals.localAnnotationsName && typeof localAnnotationsName !== 'undefined') { %>,
                 <% if (locals.localAnnotationsName && typeof localAnnotationsName !== 'undefined') { %>
-                    <% if (Array.isArray(localAnnotationsName)) { %>
+                    <% if (Array.isArray(localAnnotationsName) && localAnnotationsName.length > 0) { %>
```
Comment (IainSAP) [inline packages/odata-service-writer/templates/extend/manifest.json:60]: Any tests for these changes?

Comment (broksy) [reply packages/odata-service-writer/templates/extend/manifest.json:60]: There is https://github.com/SAP/open-ux-tools/blob/main/packages/odata-service-writer/test/unit/updates.test.ts#L153 for multiple local annotations, but I can add test for `localAnnotationsName.length === 0` case.

**Review by devinea — APPROVED** [review]
@broksy Thanks for refactoring to replace ejs template. 
Code changes look good and covered by tests. 
No expected snapshot changes. 
changeset ✅

**Review by IainSAP — APPROVED** [review]
Thanks @broksy 

- Couple of minor comments. Can be addressed in follow up PRs.
- Tests updated to cover.
- No output (snapshot) changes as expected

**Pull request merged**